code/__HELPERS/unsorted.dm

```diff
@@ -1507,3 +1507,13 @@ proc/pick_closest_path(value, list/matches = get_fancy_list_of_atom_types())
 	else if(powerused < 1000000000) //Less than a GW
 		return "[round((powerused * 0.000001),0.001)] MW"
 	return "[round((powerused * 0.000000001),0.0001)] GW"
+
+// Bucket a value within boundary
+/proc/get_bucket(var/bucket_size, var/max, var/current, var/min = 0, var/list/boundary_terms = list())
+	if (boundary_terms.len == 2)
+		if (current >= max) 
+			return boundary_terms[1]
+		if (current < min)
+			return boundary_terms[2]
+
+	return CEILING((bucket_size / max) * current, 1)
```

Comment:
This isn't equivalent to the old switch, for example if a mob has 90 health out of 100 maxhealth :
* the switch gives `hud_used.healths.icon_state = "health_15"`
* this formula gives `CEILING((17/ 100) * 90, 1) = CEILING(15,3, 1) = 16`, so it uses `"health_16"`. 

Also it can uses `"health_17"` which (i assume) doesn't exist.

Reply:
I  reduced down the buckets from 17 to 16, as you're right i didn't meant to go over 16.

The fact that the switch and proc return different results, Im fine with personally.

Reply:
> The fact that the switch and proc return different results, Im fine with personally.

Then it should be clear in the changelog that these changed (even slightly), as players may be used to the old thresholds.